packages/ui/src/models/visualization/flows/nodes/mappers/parallel-processor-base-node-mapper.ts

```diff
@@ -0,0 +1,73 @@
+import { createVisualizationNode } from '../../../visualization-node';
+import { CamelRouteVisualEntityData, ICamelElementLookupResult } from '../../support/camel-component-types';
+import { BaseNodeMapper } from './base-node-mapper';
+import { CamelComponentSchemaService } from '../../support/camel-component-schema.service';
+import { getValue } from '../../../../../utils';
+
+export abstract class ParallelProcessorBaseNodeMapper extends BaseNodeMapper {
+  abstract getProcessorName(): keyof ProcessorDefinition;
+
+  getVizNodeFromProcessor(
+    path: string,
+    _componentLookup: ICamelElementLookupResult,
+    entityDefinition: unknown,
+  ): IVisualizationNode {
+    const processorName = this.getProcessorName();
+
+    const data: CamelRouteVisualEntityData = {
+      path,
+      icon: NodeIconResolver.getIcon(processorName, NodeIconType.EIP),
+      processorName,
+      isGroup: true,
+    };
+
+    const vizNode = createVisualizationNode(processorName, data);
+    const children = this.getChildrenFromBranch(`${path}.steps`, entityDefinition);
+    children.forEach((child) => {
+      vizNode.addChild(child);
+    });
+
+    return vizNode;
+  }
+
+  getChildrenFromBranch(path: string, entityDefinition: unknown): IVisualizationNode[] {
+    const stepsList = getValue(entityDefinition, path, []) as ProcessorDefinition[];
+
+    return stepsList.reduce((accStepsNodes, step, index) => {
+      const singlePropertyName = Object.keys(step)[0];
+      const childPath = `${path}.${index}.${singlePropertyName}`;
+      const childComponentLookup = CamelComponentSchemaService.getCamelComponentLookup(
+        childPath,
+        getValue(step, singlePropertyName),
+      );
+
+      const vizNode = this.rootNodeMapper.getVizNodeFromProcessor(childPath, childComponentLookup, entityDefinition);
+      this.addNodeInteraction(vizNode, childComponentLookup.processorName as keyof ProcessorDefinition);
+      accStepsNodes.push(vizNode);
+      return accStepsNodes;
+    }, [] as IVisualizationNode[]);
+  }
+
+  addNodeInteraction(vizNode: IVisualizationNode, processorName: keyof ProcessorDefinition): void {
+    const stepsProperties = CamelComponentSchemaService.getProcessorStepsProperties(processorName);
+    const canHaveChildren = stepsProperties.find((property) => property.type === 'branch') !== undefined;
+    const canHaveSpecialChildren = Object.keys(stepsProperties).length > 1;
+    const canReplaceStep = CamelComponentSchemaService.canReplaceStep(processorName);
+    const canRemoveStep = !CamelComponentSchemaService.DISABLED_REMOVE_STEPS.includes(processorName);
+    const canBeDisabled = CamelComponentSchemaService.canBeDisabled(processorName);
+
+    vizNode.setNodeInteraction({
+      canHavePreviousStep: false,
+      canHaveNextStep: false,
+      canRemoveFlow: false,
+      canHaveChildren,
+      canHaveSpecialChildren,
+      canReplaceStep,
+      canRemoveStep,
+      canBeDisabled,
+    });
+  }
```

Comment:
This also seems very similar to the [abstract-camel-visual-entity](https://github.com/KaotoIO/kaoto/blob/972d250151e6c43a8349d9b346c44b7b67589cae/packages/ui/src/models/visualization/flows/abstract-camel-visual-entity.ts#L185-L206) counterpart.

Reply:
We can take care of this in another PR, so we can also remove the other duplication from other visual entities as well.

Reply:
https://github.com/KaotoIO/kaoto/issues/1445